server/main_server/src/rodsServer.cpp

```diff
@@ -631,8 +631,20 @@ namespace
 
             zmq::context_t zmq_ctx;
             zmq::socket_t zmq_socket{zmq_ctx, zmq::socket_type::req};
-            zmq_socket.set(zmq::sockopt::sndtimeo, 500);
-            zmq_socket.set(zmq::sockopt::rcvtimeo, 5000);
+            const int timeout = [] {
```

Comment:
Consider changing the variable name so that it's clear we're dealing with milliseconds.

Perhaps `timeout_ms`?

Or maybe we introduce a new config keyword as an alias. This new keyword would include `_IN_MILLISECONDS` or something to help readers understand the `10000`.

Reply:
Something like this?
```diff
--- a/server/main_server/src/rodsServer.cpp
+++ b/server/main_server/src/rodsServer.cpp
@@ -631,17 +631,18 @@ namespace
 
             zmq::context_t zmq_ctx;
             zmq::socket_t zmq_socket{zmq_ctx, zmq::socket_type::req};
-            const int timeout = [] {
+            const int server_control_plane_timeout_in_milliseconds = [] {
                 try {
                     const auto config_handle = irods::server_properties::instance().map();
                     return config_handle.get_json().at(irods::KW_CFG_SERVER_CONTROL_PLANE_TIMEOUT).get<int>();
                 }
                 catch (...) {
-                    return 10000;
+                    constexpr int default_server_control_plane_timeout_in_milliseconds = 10000;
+                    return default_server_control_plane_timeout_in_milliseconds;
                 }
             }();
-            zmq_socket.set(zmq::sockopt::sndtimeo, timeout);
-            zmq_socket.set(zmq::sockopt::rcvtimeo, timeout);
+            zmq_socket.set(zmq::sockopt::sndtimeo, server_control_plane_timeout_in_milliseconds);
+            zmq_socket.set(zmq::sockopt::rcvtimeo, server_control_plane_timeout_in_milliseconds);
             zmq_socket.set(zmq::sockopt::linger, 0);
             zmq_socket.connect(zmq_server_target);
             log_server::trace("Connection established.");
```

Reply:
Very clear. No one has to guess what is happening.

I think we have a file that defines global const[expr] default values. Perhaps the constexpr default can go there. Then it's available for other locations.

Sadly, I can't remember which file that is. Feel free to look for it, but if not, don't worry about it. What you've presented is certainly good enough.

Reply:
Hmm... if you can give an example for a default value that you have in mind, I may be able to locate it. So far, all default values I've found seem to be like this one - that is, local to where they are being used - or scattered throughout various header files which may or may not be related to the respective values. :)

Reply:
The thing I had in mind turned out to be specific to the delay server.

The delay server's hpp file contains constexpr variables. The function where I remember one of those being used was in a MSI that was recently removed.

All of that to say, I think you can move forward with the diff you presented.

Reply:
Okay. I say we leave it as this diff, then. Resolving